nodes/core/core/58-debug.html

```diff
@@ -119,7 +119,13 @@
 
             function getTimestamp() {
                 var d = new Date();
-                return d.toLocaleString();
+                var options = {};
+                var lang = "en-US";
+                if(RED.text.bidi.getCalendarType() === "hijri")
+                	options = lang + "-u-ca-islamic";
+                else if(RED.text.bidi.getCalendarType() === "hebrew")
+                	options = lang + "-u-ca-hebrew";
+                return d.toLocaleString(options);
             }
```

Comment:
As we're likely to have other places that need to format a Date object properly, this code should be a utility function in `bidi.js`